File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/HardeningSkin.lua
function GetName()
  return "Hardening Skin"
end

function GetType()
  return Passive
end

function GetHookPoints()
  return { TurnTriggered }
end

function OnAttachment(char)
end

function OnRemoval(char)
end

function Trigger(hookPoint, char)
  if hookPoint ~= TurnTriggered then
    return
  end

  char:BuffStat(DEF, 0.2)
  char:BuffStat(MR, 0.2)
end

File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/HardeningSkin.cs
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Afflictions
{
    public class HardeningSkin : Affliction
    {
        public override string Name { get { return "Hardening Skin"; } }

        public override AfflictionType Type { get { return AfflictionType.Passive; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new[]
                {
                    HookPoint.TurnTriggered, 
                };
            }
        }

        protected override void OnTurnTriggered(ICharacter character)
        {
            character.BuffStat(Stat.DEF, .2);
            character.BuffStat(Stat.MR, .2);
        }
    }
}